lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -294,6 +320,16 @@ def _delete_and_create_service(self, client: Client):
         client.delete(Service, self._app, namespace=self._namespace)
         client.create(service)
 
+    def _delete_and_create_new_service(self, client: Client):
+        try:
+            service = client.get(Service, self.service_name, namespace=self._namespace)
```

Comment:
Should there be a guard here against using app_name as the service name?  I see the note in the docstring above about avoiding app_name since the default juju service uses that name, but afaict nothing actually enforces that suggestion.  That sounds like an easy way for the user to make a mistake